### PR TITLE
detect and use third-party analysis backends when possible

### LINKED ISSUE #2376: detect and use third-party analysis backends when possible
capa has analysis backends for third party tools, like IDA, Binary Ninja, and Ghidra. It's possible to invoke capa to use these backends, but you typically have to be an expert to figure out how to do this.

We should investigate whether the standalone capa.exe can automatically detect installations of relevant third party applications and use their backends, when possible. For example, capa.exe could recognize that IDA is installed, figure out where the path is, and try to initialize idalib from that path. Of course, there are a lot of edge cases and possibilities, but I suspect a few common workflows might be achievable. For example, ensuring capa.exe in FLARE-VM uses the best available backend. Worst case, capa.exe continues to use the embedded vivisect backend.
```[tasklist]
### Tasks
- [x] IDA 9.0 idalib on Linux
- [ ] IDA 9.0 idalib on Windows
- [x] Binary Ninja on Linux
- [ ] Binary Ninja on Windows
- [ ] Binary Ninja on macOS
- [ ] IDA 9.0 idalib on macOS
```

### PR DESCRIPTION
**Pull request by williballenthin**
closes #2376

### Checklist
- [x] No new tests needed: don't have an idalib license we can use in CI right now
- [ ] Documentation updates

### CODE REVIEW

File: capa/loader.py [hunk 1]
```diff
@@ -321,6 +315,36 @@ def get_extractor(
...
+        if not idalib.load_idalib():
+            raise RuntimeError("failed to load IDA idalib module.")
+
+        import ida
+        import ida_auto
+
+        import capa.features.extractors.ida.extractor
+
+        logger.debug("idalib: opening database...")
+        # idalib writes to stdout (ugh), so we have to capture that
```
[the remark this answers is not in the document]
Comment (williballenthin) [reply capa/loader.py:335]: yeah I have a thread going with them with a few points of feedback, including this.

**Comment by williballenthin** [comment]
confirmed working with PyInstaller: 

image

**Comment by williballenthin** [comment]
Given that I don't have easy access to a Windows install of IDA v9.0 Beta 4, let's stick to just Linux support.

**Review by mr-tz — APPROVED** [review]
very cool!

**Comment by williballenthin** [comment]
need to clean up the IDA database (which idalib leaves strewn around as .nam, .til, .id0, .id1, etc.)

**Pull request merged**